### PR TITLE
upgrade ROCm version  to 5.0.0

### PR DESCRIPTION
**Pull request by weihanmines**
Thanks for your contribution! Unfortunately, tensorflow/runtime is currently not
accepting contributions. Please see the
[Contribution Guidelines](../blob/master/README.md#contribution-guidelines) for
more information.

@chsigg @hanbinyoon @deven-amd

### CODE REVIEW

**Comment by deven-amd** [comment]
@weihanmines 

we should also include the corresponding `third_party/hip/*.inc` files (generated from ROCm 5.0.0) as part of this PR

File: third_party/hip/hip_stub.h.inc [hunk 1]
```diff
@@ -54,6 +54,7 @@ enum hipError_t {
   hipErrorOperatingSystem = 304,
   hipErrorInvalidHandle = 400,
   hipErrorInvalidResourceHandle = 400,
+  hipErrorIllegalState = 401,
```
Comment (chsigg) [inline third_party/hip/hip_stub.h.inc:57]: Do we also need to update printing for these error enums?

Comment (chsigg) [reply third_party/hip/hip_stub.h.inc:57]: Actually, I will let you do this in a separate change.

Comment (weihanmines) [reply third_party/hip/hip_stub.h.inc:57]: > 

Probably not. However, they are added in ROCm 5.0.

**Pull request merged**